bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/CallController.java

```diff
@@ -119,17 +120,17 @@ public void disableAutoInboundFlowControl() {
   Note: The request count may not be perfectly right, if there's a situation where request(1) is called
   at the same time as an RPC failure.
    */
-  public void request(int count) {
+  public synchronized void request(int count) {
     call.request(count);
   }
 
   @Override
-  public void setMessageCompression(boolean enable) {
+  public synchronized void setMessageCompression(boolean enable) {
     call.setMessageCompression(enable);
```

Comment:
I don't think this needs to be sync because it should only be called on init. I actually think this should throw an UnsupportedException because its setting will be lost on retry